src/main/java/com/fasterxml/jackson/core/util/TextBuffer.java

```diff
@@ -694,7 +694,7 @@ public void append(char[] c, int start, int len)
         // And then allocate new segment; we are guaranteed to now
         // have enough room in segment.
         do {
-            expand(len);
+            expand();
             int amount = Math.min(_currentSegment.length, len);
             System.arraycopy(c, start, _currentSegment, 0, amount);
```

Comment:
... and here presumably.... hmmh. I guess it actually isn't strictly required that length. It's just sort of a ... wish?

(for a moment I thought this is where `len` must be obeyed but no, calc of `amount` takes into account that we get what we get).